### DISCUSSION TITLE
Integrating fzf and textadept to produce filename completions

### DISCUSSION THREAD

**Discussion by quesada** [orbitalquark/textadept#521] [Q&A]
I need filename completion, ideally with fuzzy matching, for markdown wiki-links. 
Every attempt to use a markdown lsp with textadept has failed. So I'm now thinking that doing something simpler, like using fzf to copy a filename into the clipboard and pasting it on the textadept buffer, would be easier.

Example of integration doing this in vim:
https://github.com/junegunn/fzf.vim/issues/772

Is there an easy way to do this in textadept? for the curses version, it could be as easy as sending it to background, calling fzf, and pasting the resulting filename in the open buffer? Ideally working on both gui and curses version, but curses would be ennough.

**Comment by rgieseke** [comment]
[Textredux](https://github.com/rgieseke/textredux/) has a fuzzy search list. How do you get your markdown wiki links?

**Comment by quesada** [comment]
Just to answer my own question, this is a fish shell function that achieves what I want:

```
function fzfcopy
    set file (fzf)
    if test -n "$file"
        echo -n "$file" | xclip -selection clipboard
        # Use `pbcopy` instead of `xclip` if you're on macOS
    end
end
```
